test/files/team.iced

```diff
@@ -50,3 +52,42 @@ exports.test_bad_key_section = (T,cb) ->
   T.equal err.message, "Need per_team_key.generation to be an integer > 0 (got undefined)", "right message"
   cb null
 
+round_trip_with_corrupted_reverse_sig = ({T, corrupt}, cb) ->
+  esc = make_esc cb, "test_bad_key_section"
+  await KeyManager.generate {}, esc defer km
+  arg = new_sig_arg { km }
+  arg.team = { members : { admin : ["a"] } }
+  arg.kms = {}
+  await EncKeyManager.generate {}, esc defer arg.kms.encryption
+  await KeyManager.generate {}, esc defer arg.kms.signing
+  arg.kms.generation = 10
+  obj = new team.RotateKey arg
+
+  obj._v_generate = (opts, cb) ->
+    esc = make_esc cb, "_v_generate"
+    x = { reverse_sig: null }
+    @set_new_key_section x
+    eng = @get_new_km().make_sig_eng()
+    await @generate_json { version : opts.version }, esc defer msg
+    msg2 = JSON.parse msg
+    msg2.foo = "bar" if corrupt
+    msg = json_stringify_sorted(msg2)
+    await eng.box msg, esc defer { armored, type }
+    x.reverse_sig = armored
+    cb null
+
+  await obj.generate_v2 esc defer out
+  typ = out.inner.obj.body.type
+  obj2 = alloc typ, arg
+  varg = { armored : out.armored, skip_ids : true, make_ids : true, inner : out.inner.str }
+  await obj2.verify_v2 varg, defer err
+  if corrupt
+    T.assert err?, "got an error back"
+    T.assert (err.message.indexOf('Reverse sig json mismatch') >= 0), "found right error message"
```

Comment:
Should we be asserting anything in the `!corrupt` case? It does look like we mean to test that case below.